### PR TITLE
Fix mangling of enqueue_marker

### PR DESCRIPTION
**Pull request by svenvh**
Minor refactoring to aid debugging of mangler substitution issues.

Fix a mangling issue for `enqueue_marker`.

### CODE REVIEW

File: lib/SPIRV/Mangler/Mangler.cpp [hunk 1]
```diff
@@ -92,15 +92,15 @@ class MangleVisitor : public TypeVisitor {
...
+    // Builtin primitives such as int are not substitution candidates, but
+    // all other primitives are.  Even though most of these do not appear
+    // repeatedly in builtin function signatures, we need to track them in
+    // the substitution map.
+    if (T->getPrimitive() >= PRIMITIVE_STRUCT_FIRST) {
       if (!mangleSubstitution(T, mangledPrimitiveString(T->getPrimitive()))) {
         size_t Index = Stream.str().size();
         Stream << mangledPrimitiveString(T->getPrimitive());
-        Substitutions[Stream.str().substr(Index)] = SeqId++;
+        recordSubstitution(Stream.str().substr(Index), SeqId++);
```
Comment (AlexeySachkov) [inline lib/SPIRV/Mangler/Mangler.cpp]: `recordSubstitution` is a pretty simple function which just replaces call to `operator[]` - would it make sense to move usages of `SeqId` into it too? As I can see it is not used anywhere else

File: lib/SPIRV/Mangler/Mangler.cpp [hunk 2]
```diff
@@ -187,6 +187,10 @@ class MangleVisitor : public TypeVisitor {
   }
 
 private:
+  void recordSubstitution(std::string Str, unsigned Id) {
```
Comment (AlexeySachkov) [inline lib/SPIRV/Mangler/Mangler.cpp]: Why not `const std::string &Str`?

File: lib/SPIRV/Mangler/Mangler.cpp (hunk 1 above)
Comment (svenvh) [reply lib/SPIRV/Mangler/Mangler.cpp]: Good point, done!

File: lib/SPIRV/Mangler/Mangler.cpp (hunk 2 above)
Comment (svenvh) [reply lib/SPIRV/Mangler/Mangler.cpp]: Yeah why not, done! :)

**Pull request merged**